File: push/src/main/java/com/xiaomi/smack/XMPPException.java
package com.xiaomi.smack;

import com.xiaomi.smack.packet.StreamError;
import com.xiaomi.smack.packet.XMPPError;
import java.io.PrintStream;
import java.io.PrintWriter;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/smack/XMPPException.class */
public class XMPPException extends Exception {
    private XMPPError error;
    private StreamError streamError;
    private Throwable wrappedThrowable;

    public XMPPException() {
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
    }

    public XMPPException(StreamError streamError) {
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
        this.streamError = streamError;
    }

    public XMPPException(XMPPError xMPPError) {
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
        this.error = xMPPError;
    }

    public XMPPException(String str) {
        super(str);
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
    }

    public XMPPException(String str, XMPPError xMPPError) {
        super(str);
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
        this.error = xMPPError;
    }

    public XMPPException(String str, XMPPError xMPPError, Throwable th) {
        super(str);
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
        this.error = xMPPError;
        this.wrappedThrowable = th;
    }

    public XMPPException(String str, Throwable th) {
        super(str);
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
        this.wrappedThrowable = th;
    }

    public XMPPException(Throwable th) {
        this.streamError = null;
        this.error = null;
        this.wrappedThrowable = null;
        this.wrappedThrowable = th;
    }

    @Override // java.lang.Throwable
    public String getMessage() {
        StreamError streamError;
        XMPPError xMPPError;
        String message = super.getMessage();
        return (message != null || (xMPPError = this.error) == null) ? (message != null || (streamError = this.streamError) == null) ? message : streamError.toString() : xMPPError.toString();
    }

    public StreamError getStreamError() {
        return this.streamError;
    }

    public Throwable getWrappedThrowable() {
        return this.wrappedThrowable;
    }

    public XMPPError getXMPPError() {
        return this.error;
    }

    @Override // java.lang.Throwable
    public void printStackTrace() {
        printStackTrace(System.err);
    }

    @Override // java.lang.Throwable
    public void printStackTrace(PrintStream printStream) {
        super.printStackTrace(printStream);
        if (this.wrappedThrowable != null) {
            printStream.println("Nested Exception: ");
            this.wrappedThrowable.printStackTrace(printStream);
        }
    }

    @Override // java.lang.Throwable
    public void printStackTrace(PrintWriter printWriter) {
        super.printStackTrace(printWriter);
        if (this.wrappedThrowable != null) {
            printWriter.println("Nested Exception: ");
            this.wrappedThrowable.printStackTrace(printWriter);
        }
    }

    @Override // java.lang.Throwable
    public String toString() {
        StringBuilder sb = new StringBuilder();
        String message = super.getMessage();
        if (message != null) {
            sb.append(message);
            sb.append(": ");
        }
        XMPPError xMPPError = this.error;
        if (xMPPError != null) {
            sb.append(xMPPError);
        }
        StreamError streamError = this.streamError;
        if (streamError != null) {
            sb.append(streamError);
        }
        if (this.wrappedThrowable != null) {
            sb.append("\n  -- caused by: ");
            sb.append(this.wrappedThrowable);
        }
        return sb.toString();
    }
}

File: legacy/src/main/java/com/xiaomi/smack/XMPPException.kt
package com.xiaomi.smack

import com.xiaomi.smack.packet.StreamError
import com.xiaomi.smack.packet.XMPPError
import java.io.PrintStream
import java.io.PrintWriter

/*
 * Stock reference: com.xiaomi.xmsf 7.4.67-C (versionCode 70004067),
 * split-XiaomiServiceFrameworkCN-master.apk sha256 444e9f128591e04e38672bfe44a246ab3fa97ae68e95882839d8a7afe766df2b,
 * JADX path: com.xiaomi.xmsf/stock/split-XiaomiServiceFrameworkCN-master/sources/qa/j.java
 * Stock class name is obfuscated as qa.j; this file keeps the deobfuscated com.xiaomi.smack.XMPPException API.
 */
class XMPPException : Exception {
    var error: XMPPError? = null
        private set
    var streamError: StreamError? = null
        private set
    var wrappedThrowable: Throwable? = null
        private set

    constructor()

    constructor(streamError: StreamError) {
        this.streamError = streamError
    }

    constructor(error: XMPPError) {
        this.error = error
    }

    constructor(message: String) : super(message)

    constructor(
        message: String,
        error: XMPPError,
    ) : super(message) {
        this.error = error
    }

    constructor(
        message: String,
        error: XMPPError,
        wrappedThrowable: Throwable,
    ) : super(message) {
        this.error = error
        this.wrappedThrowable = wrappedThrowable
    }

    constructor(
        message: String,
        wrappedThrowable: Throwable,
    ) : super(message) {
        this.wrappedThrowable = wrappedThrowable
    }

    constructor(wrappedThrowable: Throwable) {
        this.wrappedThrowable = wrappedThrowable
    }

    override val message: String?
        get() {
            val superMessage = super.message
            return when {
                superMessage != null -> superMessage
                error != null -> error.toString()
                streamError != null -> streamError.toString()
                else -> null
            }
        }

    fun getXMPPError(): XMPPError? = error

    override fun printStackTrace() {
        printStackTrace(System.err)
    }

    override fun printStackTrace(s: PrintStream) {
        super.printStackTrace(s)
        wrappedThrowable?.let {
            s.println("Nested Exception: ")
            it.printStackTrace(s)
        }
    }

    override fun printStackTrace(s: PrintWriter) {
        super.printStackTrace(s)
        wrappedThrowable?.let {
            s.println("Nested Exception: ")
            it.printStackTrace(s)
        }
    }

    override fun toString(): String {
        val sb = StringBuilder()
        super.message?.let {
            sb.append(it)
            sb.append(": ")
        }
        error?.let { sb.append(it) }
        streamError?.let { sb.append(it) }
        wrappedThrowable?.let {
            sb.append("\n  -- caused by: ")
            sb.append(it)
        }
        return sb.toString()
    }
}
